### PR TITLE
MSAL Angular SPA

### PR DESCRIPTION
**Pull request by skabou**
Initial import DOCS-CODE-0022

### CODE REVIEW

**Comment by skabou** [comment]
This should look familiar to you 😄  ckittel

**Review by ckittel — APPROVED** [review]
LGTM (again ;)), good for CD.

**Comment by ckittel** [comment]
@skabou , I added @mmacy as a reviewer since you mentioned in standup this morning that this was ready for review by him.  Please adjust expectations if that's not the case.

**Comment by skabou** [comment]
Thanks @ckittel !

File: angular-spa/README.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+Follow the instructions that are presented on the web browser. If everything worked, you should receive a response similar to this on the "View Profile" page:
+
+![Screenshot with profile information from Graph](profile.png?raw=true)
+
+## About the code
+
+This Angular application uses the Angular MSAL v2 library to authenticate a user and then makes a request to the Graph API as the authenticated user.
+- MSALGuard is used to protect routes
+- MSALBroadcastService is used to consume events emitted by MSAL
```
Comment (mmacy) [inline angular-spa/README.md:86]: Which event(s) and why?

File: angular-spa/README.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+- MSALInterceptor is provided by MSAL Angular to automatically acquire tokens for protected resources
```
Comment (mmacy) [inline angular-spa/README.md:87]: Access tokens, ID tokens, or both?

File: angular-spa/README.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+- HttpClient is used to perform an HTTP GET request against Graph with the user's token
```
Comment (mmacy) [inline angular-spa/README.md:88]:
```suggestion
- HttpClient is used to issue an HTTP GET request to Microsoft Graph with user's access token.
```

File: angular-spa/README.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+ng serve --open
+```
+
+Follow the instructions that are presented on the web browser. If everything worked, you should receive a response similar to this on the "View Profile" page:
+
+![Screenshot with profile information from Graph](profile.png?raw=true)
+
+## About the code
+
+This Angular application uses the Angular MSAL v2 library to authenticate a user and then makes a request to the Graph API as the authenticated user.
```
Comment (mmacy) [inline angular-spa/README.md:84]:
```suggestion
This Angular web application uses the Angular MSAL v2 library to authenticate a user and then makes a request to the Microsoft Graph API as the authenticated user to retrieve their user profile data.
```

File: angular-spa/README.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+```console
+ng serve --open
+```
+
+Follow the instructions that are presented on the web browser. If everything worked, you should receive a response similar to this on the "View Profile" page:
+
+![Screenshot with profile information from Graph](profile.png?raw=true)
+
+## About the code
```
Comment (mmacy) [inline angular-spa/README.md:82]: Please note the filenames each of these are used in.

File: angular-spa/README.md [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+
+This Angular application uses the Angular MSAL v2 library to authenticate a user and then makes a request to the Graph API as the authenticated user.
+- MSALGuard is used to protect routes
```
Comment (mmacy) [inline angular-spa/README.md:85]:
```suggestion
- MSALGuard is used to protect the web app's routes.
```

Also, which routes?

**Review by mmacy — CHANGES_REQUESTED** [review]
@skabou Thanks, Jason. A few changes (clarifications, mostly) requested.

**Comment by skabou** [comment]
ok round 2... Think I covered everything.  Please review when ready! @mmacy

**Pull request merged**